addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -0,0 +1,314 @@
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_id_efaktur_id = fields.Many2one('l10n_id_efaktur.efaktur', string="E-faktur", copy=False)
+    l10n_id_tax_number = fields.Char(string="Tax Number", inverse='_inverse_tax_number', compute='_compute_tax_number', store=False, copy=False)
+    l10n_id_replace_invoice_id = fields.Many2one('account.move', string="Replace Invoice",  domain="['|', '&', '&', ('state', '=', 'posted'), ('partner_id', '=', partner_id), ('reversal_move_id', '!=', False), ('state', '=', 'cancel')]", copy=False)
+    l10n_id_upload_user = fields.Many2one('res.users', 'Upload User', readonly=True, copy=False)
+    l10n_id_attachment_id = fields.Many2one('ir.attachment', readonly=True, copy=False)
+    l10n_id_date_validated = fields.Datetime('Upload Date', readonly=True, copy=False)
+    l10n_id_csv_created = fields.Boolean('CSV Created', compute='_compute_csv_created', copy=False)
+    l10n_id_kode_transaksi = fields.Selection([
+            ('01', '01 Kepada Pihak yang Bukan Pemungut PPN (Customer Biasa)'),
+            ('02', '02 Kepada Pemungut Bendaharawan (Dinas Kepemerintahan)'),
+            ('03', '03 Kepada Pemungut Selain Bendaharawan (BUMN)'),
+            ('04', '04 DPP Nilai Lain (PPN 1%)'),
+            ('06', '06 Penyerahan Lainnya (Turis Asing)'),
+            ('07', '07 Penyerahan yang PPN-nya Tidak Dipungut (Kawasan Ekonomi Khusus/ Batam)'),
+            ('08', '08 Penyerahan yang PPN-nya Dibebaskan (Impor Barang Tertentu)'),
+            ('09', '09 Penyerahan Aktiva ( Pasal 16D UU PPN )'),
+        ], string='Kode Transaksi', help='Dua digit pertama nomor pajak',
+        readonly=True, states={'draft': [('readonly', False)]}, copy=False)
+    l10n_id_need_kode_transaksi = fields.Boolean(compute='_compute_need_kode_transaksi')
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        self.l10n_id_kode_transaksi = self.partner_id.l10n_id_kode_transaksi
+        return super(AccountMove, self)._onchange_partner_id()
+
+    def _inverse_tax_number(self):
+        for record in self:
+            if record.type == 'in_invoice':
+                if record.l10n_id_efaktur_id:
+                    record.l10n_id_efaktur_id.name = record.l10n_id_tax_number
+                elif record.l10n_id_tax_number:
+                    self.env['l10n_id_efaktur.efaktur'].create({
+                        'name': record.l10n_id_tax_number,
+                        'invoice_id': [(6, 0, record.ids)],
+                        'company_id': record.company_id.id,
+                    })
+            elif record.l10n_id_efaktur_id.name != record.l10n_id_tax_number:
+                raise UserError(_("You can't change the number manually for a Customer Invoice"))
+
+    @api.depends('l10n_id_efaktur_id')
+    def _compute_tax_number(self):
+        for record in self:
+            record.l10n_id_tax_number = record.l10n_id_efaktur_id.name
```

Comment:
could be a related and it is not even needed as the efaktur object one2one makes no sense

Reply:
It is computed because there is an inverse function, it is for usability in the vendor bills